src/lib/node_error_service/node_error_service.ml

```diff
@@ -0,0 +1,144 @@
+open Async
+open Core
+open Pipe_lib
+open Signature_lib
+
+type node_error_data =
+  { peer_id : string
+  ; ip_address : string
+  ; public_key : Public_key.Compressed.t option
+  ; git_branch : string
+  ; commit_hash : string
+  ; chain_id : string
+  ; contact_info : string option
+  ; timestamp : string
+  ; id : string
+  ; error : string
+  ; catchup_job_states : Transition_frontier.Full_catchup_tree.job_states option
```

Comment:
Nit: use the `Node_status_service.catchup_job_states` type when reinstated, or better still, re-alias it here as below.